R/codemeta_readme.R

```diff
@@ -1,153 +1,148 @@
 
-}
+  badges <- get_badge_links_matching(readme, url)
 
-# find the readme
+  if (is.null(badges)) {
 
-guess_readme_url <- function(root){
-  if (uses_git(root)) {
-    github <- guess_github(root)
-    github <- gsub(".*com\\/", "", github)
-    github <- strsplit(github, "/")[[1]]
-    readme <- try(gh::gh("GET /repos/:owner/:repo/readme",
-                         owner = github[1], repo = github[2]),
-                  silent = TRUE)
-    if(inherits(readme, "try-error")){
-      readme_url <- NULL
-    }else{
-      readme$html_url
-    }
-
-
-  } else{
-    NULL
+    return(NULL)
   }
-}
 
-guess_readme_path <- function(root){
-  readmes <- dir(root, pattern = "^README\\.R?md$", ignore.case = TRUE)
-  if(length(readmes) == 0){
-    readme_path <- NULL
-  }else{
-    readme_rmd <- readmes[grepl("\\.[Rr]md", readmes)]
-    if(length(readme_rmd) == 0){
-      readme_path <- file.path(root, readmes[grepl("\\.md", readmes)])
-    }else{
-      readme_path <- file.path(root, readme_rmd)
-    }
-  }
+  review <- as.numeric(stringr::str_remove(badges, paste0(".*https://", url)))
 
-  if(length(readme_path) > 1) { # README.md and ReadMe.md could both exist ...
-    ## silently use the first match (locale-dependent)
-    readme_path <- readme_path[1]
-  }
+  if (review %in% ropensci_reviews()$review) {
 
-  return(readme_path)
+    list("@type" = "Review",
+         "url" = paste0("https://", url, review),
+         "provider" = "http://ropensci.org")
+  }
+  # else NULL implicitly
 }
 
-.guess_readme <- function(root = ".") {
+# guess_readme_url -------------------------------------------------------------
+# find the readme
+guess_readme_url <- function(root) {
 
-  readme_url <- guess_readme_url(root)
+  if (! uses_git(root)) {
 
-  readme_path <- guess_readme_path(root)
+    return(NULL)
+  }
 
-  return(list(readme_path = readme_path,
-              readme_url = readme_url))
+  github <- stringr::str_remove(guess_github(root), ".*com\\/")
 
-}
+  parts <- strsplit(github, "/")[[1]]
 
+  readme <- try(silent = TRUE, gh::gh(
+    "GET /repos/:owner/:repo/readme", owner = parts[1], repo = parts[2]
+  ))
 
-guess_readme <- memoise::memoise(.guess_readme)
+  if (! inherits(readme, "try-error")) {
 
+    readme$html_url
+  }
+  # else NULL implicitly
+}
 
+# guess_readme_path ------------------------------------------------------------
+guess_readme_path <- function(root) {
 
+  readmes <- dir(root, pattern = "^README\\.R?md$", ignore.case = TRUE)
+
+  if (length(readmes) == 0) {
 
-codemeta_readme <- function(readme, codemeta){
-  if (is.null(codemeta$contIntegration)){
-    codemeta$contIntegration <- guess_ci(readme)
+    return(NULL)
   }
 
-  if (is.null(codemeta$developmentStatus)){
-    codemeta$developmentStatus <-
-      guess_devStatus(readme)
+  readme_rmd <- grep("\\.[Rr]md$", readmes, value = TRUE)
+
+  if (length(readme_rmd) == 0) {
+
+    readme_file <- grep("\\.md$", readmes, value = TRUE)
```

Comment:
`readme_file` does not get a value if `length(readme_rmd) != 0` which I think is the reason why one test is failing.

Reply:
You are right, my fault, sorry. I am fixing this.

Reply:
no problem and no rush :-)